### ISSUE TITLE
allow null byte between syslines

### ISSUE THREAD

**Issue by jtmoon79** [jtmoon79/super-speedy-syslog-searcher#80]
tl;dr allow user to have arbitrary "record separator", like null byte, between syslines

### Problem

While writing an [SO Answer about converting plain syslog files to journal log entries](https://unix.stackexchange.com/questions/727837/how-can-i-use-journalctl-to-time-filter-and-view-non-journal-files/734044#734044), trying to operate on a "line by line" basis from `s4` output would produce wrong output for multi-line syslines.

Given log file `alternatives.log.1` with content

```text
update-alternatives 2023-01-13 06:12:20: run with --install /usr/bin/rview rview /usr/bin/vim.basic 30
update-alternatives 2023-01-13 06:12:20: run with --install /usr/bin/vi vi /usr/bin/vim.basic 30
--slave /usr/share/man/da/man1/vi.1.gz vi.da.1.gz /usr/share/man/da/man1/vim.1.gz
--slave /usr/share/man/de/man1/vi.1.gz vi.de.1.gz /usr/share/man/de/man1/vim.1.gz
--slave /usr/share/man/fr/man1/vi.1.gz vi.fr.1.gz /usr/share/man/fr/man1/vim.1.gz
--slave /usr/share/man/it/man1/vi.1.gz vi.it.1.gz /usr/share/man/it/man1/vim.1.gz
--slave /usr/share/man/ja/man1/vi.1.gz vi.ja.1.gz /usr/share/man/ja/man1/vim.1.gz
--slave /usr/share/man/pl/man1/vi.1.gz vi.pl.1.gz /usr/share/man/pl/man1/vim.1.gz
--slave /usr/share/man/ru/man1/vi.1.gz vi.ru.1.gz /usr/share/man/ru/man1/vim.1.gz
--slave /usr/share/man/man1/vi.1.gz vi.1.gz /usr/share/man/man1/vim.1.gz
```

The second log line ("sysline" in s4 parlance) occupies several actual text lines.

Attempting to use `s4` as a record separator will fail because using the default newline character `'\n'` as the record separator will not work in this case.

### Solution

Allow the user provide some arbitrary string as a record separator.
This will allow using `s4` along with tools like `xargs --null`.
